From 51ed078d968f53d36dea4511324e899de49e4f3a Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Wed, 17 Jul 2024 08:52:32 +0200
Subject: [PATCH 1156/1695] drivers: clk: fix stm32mp13 round_rate on PLLs max
 divisor

Fix stm32mp13 clock driver round-rate operation handler to treat
max divisor that is the divisor IOMEM cell max value + 1, e.g. 128
on a 7 bit width divisor cell, not 127.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git b8cd7102fc49ba284b8c2755b5bd89fe1f78c408 drivers: clk: fix stm32mp13 round_rate on PLLs max divisor]
Change-Id: Iec64b22803c864371130bec0f80efb5011d8e65a
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/393704
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Gabriel FERNANDEZ <gabriel.fernandez@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/drivers/clk/clk-stm32mp13.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/drivers/clk/clk-stm32mp13.c b/core/drivers/clk/clk-stm32mp13.c
index 4c132dda3..2e23c658f 100644
--- a/core/drivers/clk/clk-stm32mp13.c
+++ b/core/drivers/clk/clk-stm32mp13.c
@@ -1968,7 +1968,7 @@ static unsigned long clk_stm32_composite_round_rate(struct clk *clk,
 		return 0UL;
 
 	div = MIN(UDIV_ROUND_NEAREST((uint64_t)prate, rate),
-		  MASK_WIDTH_SHIFT(divider->width, 0));
+		  MASK_WIDTH_SHIFT(divider->width, 0) + 1);
 
 	return UDIV_ROUND_NEAREST((uint64_t)prate, div);
 }
-- 
2.47.2

